test/common/network/lc_trie_test.cc

```diff
@@ -0,0 +1,105 @@
+#include "common/common/utility.h"
+#include "common/network/address_impl.h"
+#include "common/network/cidr_range.h"
+#include "common/network/lc_trie.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Network {
+namespace LcTrie {
+typedef std::pair<std::string, std::vector<Address::CidrRange>> IpTagData;
+
+namespace {
+
+std::vector<IpTagData>
+generateTestCase(const std::vector<std::vector<std::string>>& cidr_range_strings) {
+  std::vector<std::pair<std::string, std::vector<Address::CidrRange>>> output;
+  IpTagData ip_tags;
+  for (size_t i = 0; i < cidr_range_strings.size(); i++) {
+    ip_tags.first = fmt::format("tag_{0}", i);
+    for (size_t j = 0; j < cidr_range_strings[i].size(); j++) {
+      ip_tags.second.push_back(Address::CidrRange::create(cidr_range_strings[i][j]));
+    }
+    output.push_back(ip_tags);
+    ip_tags.second.clear();
+  }
+  return output;
+};
+} // namespace
+
+// TODO(ccaraman): Add a performance benchmark test.
+
+TEST(LcTrie, Ipv4) {
```

Comment:
I'm going to restructure these tests. There is lots of duplicate in logic, take in a list of strings for cidr range, build trie, search for an input.